### PR TITLE
PR to fix scale_12_OCS worker nodes OSD

### PR DESCRIPTION
**Pull request by ramkiperiy**
Updated code to increase the OSD creation timeout
Removed assert so that exception sets flag to skip other test execution

Signed-off-by: Ramakrishnan <rperiyas@redhat.com>

### CODE REVIEW

**Review by pintojoy — APPROVED** [review]
LGTM

File: tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py [hunk 1]
```diff
@@ -75,15 +75,16 @@ def test_scale_node_and_capacity(self):
                 scale_worker_count = expected_worker_count - len(
                     existing_ocs_worker_list
                 )
-                assert scale_lib.scale_ocs_node(node_count=scale_worker_count)
+                scale_lib.scale_ocs_node(node_count=scale_worker_count)
```
Comment (keesturam) [inline tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py:78]: why remove the assert. would it work for all platforms and scenarios?

File: tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py [hunk 2]
```diff
@@ -75,15 +75,16 @@ def test_scale_node_and_capacity(self):
...
 
             # Check existing OSD count and add OSDs if required
             if existing_deviceset_count < expected_deviceset_count:
-                add_deviceset_count = (
+                add_deviceset_count = int(
```
Comment (keesturam) [inline tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py:82]: can you pls change this variable name to improve readability in line#86

File: tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py (hunk 1 above)
Comment (ramkiperiy) [reply tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py:78]: Removed assert to handle the failure in exception. With assert, exception code is not reachable. In exception code there is a flag `TestAddNode.skip_all = True` which is used in all the testcase to decide either to skip the test or not. 
i.e. In-case of failures in test_scale_node_and_capacity() testcase, exception will set `TestAddNode.skip_all = True` and following tests will be skipped.

File: tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py (hunk 2 above)
Comment (ramkiperiy) [reply tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py:82]: @keesturam I have below example part of the function(scale_capacity_with_deviceset) call https://github.com/red-hat-storage/ocs-ci/blob/742dd680cfc7b2acd4fca3103f71e04a64a599ea/ocs_ci/ocs/scale_lib.py#L1562
`    Example: If add_deviceset_count = 2 & existing storagecluster
    has deviceset count as 1, at the end of function deviceset value
    will be existing value + add_deviceset_count i.e. 3`

**Review by Avilir — APPROVED** [review]
LGTM

File: tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py (hunk 1 above)
Comment (keesturam) [reply tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py:78]: Makes sense. Thank you for the explanation Ram.

File: tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py (hunk 2 above)
Comment (keesturam) [reply tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py:82]: maybe `additional_deviceset`?

**Comment by ramkiperiy** [comment]
@vasukulkarni Updated the code to capture the exception, validation of test done locally, please verify below. Will soon push the updated code.

```
------------------------------------------------------------------------------ live log call -------------------------------------------------------------------------------
21:43:29 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Node  --selector=node-role.kubernetes.io/worker -o yaml
21:43:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get storagecluster  -n openshift-storage -o yaml
21:43:34 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get storagecluster  -n openshift-storage -o yaml
21:43:36 - MainThread - root - INFO - Due to Exception set TestAddNode.skip_all to True
21:43:36 - MainThread - root - ERROR - Cluster not in expected state. OCS worker nodes scaling Failed
PASSED                                                                                                                                                               [  9%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_scale_pvcs_pods SKIPPED                                                           [ 18%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mgr] SKIPPED                                                     [ 27%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mon] SKIPPED                                                     [ 36%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[osd] SKIPPED                                                     [ 45%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mds] SKIPPED                                                     [ 54%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[cephfsplugin] SKIPPED                                            [ 63%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[rbdplugin] SKIPPED                                               [ 72%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_rolling_reboot_node[master] SKIPPED                                               [ 81%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_rolling_reboot_node[worker] SKIPPED                                               [ 90%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_add_node_cleanup SKIPPED                                                          [100%]

====================================================================== 1 passed, 10 skipped in 13.61s ======================================================================


------------------------------------------------------------------------------ live log call -------------------------------------------------------------------------------
21:45:45 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Node  --selector=node-role.kubernetes.io/worker -o yaml
21:45:56 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get storagecluster  -n openshift-storage -o yaml
21:45:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get storagecluster  -n openshift-storage -o yaml
21:45:59 - MainThread - root - WARNING - Due to Exception set TestAddNode.skip_all to True
21:45:59 - MainThread - root - ERROR - Cluster not in expected state. Scaling OSDs Failed
PASSED                                                                                                                                                               [  9%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_scale_pvcs_pods SKIPPED                                                           [ 18%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mgr] SKIPPED                                                     [ 27%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mon] SKIPPED                                                     [ 36%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[osd] SKIPPED                                                     [ 45%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mds] SKIPPED                                                     [ 54%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[cephfsplugin] SKIPPED                                            [ 63%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[rbdplugin] SKIPPED                                               [ 72%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_rolling_reboot_node[master] SKIPPED                                               [ 81%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_rolling_reboot_node[worker] SKIPPED                                               [ 90%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_add_node_cleanup SKIPPED                                                          [100%]

====================================================================== 1 passed, 10 skipped in 26.85s ======================================================================


------------------------------------------------------------------------------ live log call -------------------------------------------------------------------------------
21:31:25 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Node  --selector=node-role.kubernetes.io/worker -o yaml
21:32:21 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get storagecluster  -n openshift-storage -o yaml
21:33:54 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: error: the server doesn't have a resource type "storagecluster"

21:33:54 - MainThread - ocs_ci.ocs.ocp - WARNING - Failed to get resource:  of kind: storagecluster, selector: None, Error: Error during execution of command: oc -n openshift-storage get storagecluster  -n openshift-storage -o yaml.
Error is error: the server doesn't have a resource type "storagecluster"

21:33:54 - MainThread - ocs_ci.ocs.ocp - WARNING - Number of attempts to get resource reached!
21:33:54 - MainThread - root - INFO - Due to Exception set TestAddNode.skip_all to True
21:33:54 - MainThread - root - ERROR - Cluster not in expected state. Error during execution of command: oc -n openshift-storage get storagecluster  -n openshift-storage -o yaml.
Error is error: the server doesn't have a resource type "storagecluster"

PASSED                                                                                                                                                               [  9%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_scale_pvcs_pods SKIPPED                                                           [ 18%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mgr] SKIPPED                                                     [ 27%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mon] SKIPPED                                                     [ 36%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[osd] SKIPPED                                                     [ 45%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mds] SKIPPED                                                     [ 54%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[cephfsplugin] SKIPPED                                            [ 63%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[rbdplugin] SKIPPED                                               [ 72%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_rolling_reboot_node[master] SKIPPED                                               [ 81%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_rolling_reboot_node[worker] SKIPPED                                               [ 90%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_add_node_cleanup SKIPPED                                                          [100%]

================================================================ 1 passed, 10 skipped in 155.47s (0:02:35) =================================================================


------------------------------------------------------------------------------ live log call -------------------------------------------------------------------------------
21:29:43 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Node  --selector=node-role.kubernetes.io/worker -o yaml
21:30:13 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: Unable to connect to the server: dial tcp 10.70.40.254:6443: i/o timeout

21:30:13 - MainThread - ocs_ci.ocs.ocp - WARNING - Failed to get resource:  of kind: Node, selector: node-role.kubernetes.io/worker, Error: Error during execution of command: oc get Node  --selector=node-role.kubernetes.io/worker -o yaml.
Error is Unable to connect to the server: dial tcp 10.70.40.254:6443: i/o timeout

21:30:13 - MainThread - ocs_ci.ocs.ocp - WARNING - Number of attempts to get resource reached!
21:30:13 - MainThread - root - INFO - Due to Exception set TestAddNode.skip_all to True
21:30:13 - MainThread - root - ERROR - Cluster not in expected state. Error during execution of command: oc get Node  --selector=node-role.kubernetes.io/worker -o yaml.
Error is Unable to connect to the server: dial tcp 10.70.40.254:6443: i/o timeout

PASSED                                                                                                                                                               [  9%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_scale_pvcs_pods SKIPPED                                                           [ 18%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mgr] SKIPPED                                                     [ 27%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mon] SKIPPED                                                     [ 36%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[osd] SKIPPED                                                     [ 45%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[mds] SKIPPED                                                     [ 54%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[cephfsplugin] SKIPPED                                            [ 63%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_respin_ceph_pods[rbdplugin] SKIPPED                                               [ 72%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_rolling_reboot_node[master] SKIPPED                                               [ 81%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_rolling_reboot_node[worker] SKIPPED                                               [ 90%]
tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py::TestAddNode::test_add_node_cleanup SKIPPED                                                          [100%]

================================================================= 1 passed, 10 skipped in 61.37s (0:01:01) =================================================================

```

File: tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py (hunk 2 above)
Comment (ramkiperiy) [reply tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py]: updated the code.

**Comment by ramkiperiy** [comment]
@keesturam @vasukulkarni Please review this PR, have made the requested changes and updated the Code verification logs in comments.

**Review by vasukulkarni — APPROVED** [review]

**Review by tiffanyn108 — APPROVED** [review]
LGTM.

**Pull request merged**